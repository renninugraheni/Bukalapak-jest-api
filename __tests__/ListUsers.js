const epListUsers = require('../endpoints/epListUsers');

describe('test api list users', () => {
    test('user can get list users at page 2',async () => {
        const queryParam = {
            page: 2,
        };
        const response = await epListUsers.epListUsers(queryParam);
        expect(response.status).toEqual(200);
        expect(response.body).toMatchSnapshot();
    });
});