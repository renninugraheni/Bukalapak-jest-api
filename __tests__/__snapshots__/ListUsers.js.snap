// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test api list users user can get list users at page 2 1`] = `
Object {
  "data": Array [
    Object {
      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/follettkyle/128.jpg",
      "email": "michael.lawson@reqres.in",
      "first_name": "Michael",
      "id": 7,
      "last_name": "Lawson",
    },
    Object {
      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/araa3185/128.jpg",
      "email": "lindsay.ferguson@reqres.in",
      "first_name": "Lindsay",
      "id": 8,
      "last_name": "Ferguson",
    },
    Object {
      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/vivekprvr/128.jpg",
      "email": "tobias.funke@reqres.in",
      "first_name": "Tobias",
      "id": 9,
      "last_name": "Funke",
    },
    Object {
      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/russoedu/128.jpg",
      "email": "byron.fields@reqres.in",
      "first_name": "Byron",
      "id": 10,
      "last_name": "Fields",
    },
    Object {
      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/mrmoiree/128.jpg",
      "email": "george.edwards@reqres.in",
      "first_name": "George",
      "id": 11,
      "last_name": "Edwards",
    },
    Object {
      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/hebertialmeida/128.jpg",
      "email": "rachel.howell@reqres.in",
      "first_name": "Rachel",
      "id": 12,
      "last_name": "Howell",
    },
  ],
  "page": 2,
  "per_page": 6,
  "total": 12,
  "total_pages": 2,
}
`;
